swap_meet/vendor.py

```diff
@@ -1,2 +1,65 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        try: 
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+        #  ValueError will be raised if item not in self.inventory
+
+    def get_by_category(self, category):
+        return [item for item in self.inventory if item.category == category]
+
+    def swap_items(self, other, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other.inventory:
+            return False    
+
+        self.add(their_item)
+        other.add(my_item)
+        self.remove(my_item)
+        other.remove(their_item)
+        return True
+    
+    def swap_first_item(self, other):
+        if not self.inventory or not other.inventory:
+            return False
+
+        return self.swap_items(other, self.inventory[0], other.inventory[0])
```

Comment:
Here, we could still go with a more error-handling approach.

```python
    def swap_first_item(self, other):
        try:
            return self.swap_items(other, self.inventory[0], other.inventory[0])
        except IndexError:
            return False
    
```

Reply:
Notice that due to the remove operations in `swap_items`, it is a linear operation, regardless of where in the list each item is. We could achieve constant time here either by swapping the items directly, or by doing an `index` then swap approach in `swap_items`.